lib/psych/class_loader.rb

```diff
@@ -2,6 +2,8 @@
 require_relative 'omap'
 require_relative 'set'
 
+autoload :Date, "date"
```

Comment:
I don't think we should set consts on Object like that.

Reply:
Would you rather change it to a `require 'date'` statement at the top-level?